### ISSUE TITLE
misuse of transient in btset tests

### ISSUE THREAD

**Issue by mpenet** [tonsky/datascript#288]
Hi,

I stumbled upon this in btset tests while reading the code: 

```
(defn into-via-doseq [to from]
  (let [res (transient [])]
    (doseq [x from]  ;; checking chunked iter
      (conj! res x))
    (persistent! res)))
```
It's a bit weird for 2 reasons:

* `to` is never used, I guess the intent was to use it as a source to transient, something like `(empty to)`,

* while this might work because of an implementation detail of vectors, this is a misuse of transient as stated by the doc. It probably doesn't matter much since it's in the tests, but you can simply replace that with (into [] ..) which is both faster (from what I tested) and correct.

> Transients support a parallel set of 'changing' operations, with similar names followed by ! - assoc!, conj! etc. These do the same things as their persistent counterparts except the return values are themselves transient. Note in particular that **transients are not designed to be bashed in-place. You must capture and use the return value in the next call**. In this way, they support the same code structure as the functional persistent code they replace. As the example will show, this will allow you to easily enhance the performance of a piece of code without structural change.

In short if you really want to go down that path you *should* use loop recur or reduce, or given the intent (realizing a potential seq, just call `doall` on it, instead of using this function.
